src/jit/assertionprop.cpp

```diff
@@ -1973,36 +1963,36 @@ AssertionInfo Compiler::optAssertionGenJtrue(GenTree* tree)
     {
         return NO_ASSERTION_INDEX;
     }
-    if (op1->gtCall.gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFINTERFACE) &&
-        op1->gtCall.gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFARRAY) &&
-        op1->gtCall.gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFCLASS) &&
-        op1->gtCall.gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFANY))
+
+    GenTreeCall* call = op1->AsCall();
+
+    if (call->gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFINTERFACE) &&
+        call->gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFARRAY) &&
+        call->gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFCLASS) &&
+        call->gtCallMethHnd != eeFindHelper(CORINFO_HELP_ISINSTANCEOFANY))
     {
         return NO_ASSERTION_INDEX;
     }
 
-    op2 = op1->gtCall.gtCallLateArgs->gtOp.gtOp2;
-    op1 = op1->gtCall.gtCallLateArgs;
+    op1 = call->gtCallLateArgs->GetNode();
+    op2 = call->gtCallLateArgs->GetNext()->GetNode();
 
     // For the assertion, ensure op1 is the object being tested.
     // Morph may have swizzled the operand order.
-    GenTree* op1op = op1->gtOp.gtOp1;
-
-    if (op1op->TypeGet() == TYP_I_IMPL)
+    if (op1->TypeGet() == TYP_I_IMPL)
     {
         jitstd::swap(op1, op2);
-        op1op = op1->gtOp.gtOp1;
     }
 
-    assert(op1op->TypeGet() == TYP_REF);
+    assert(op1->TypeGet() == TYP_REF);
 
     // Reverse the assertion
-    assert(assertionKind == OAK_EQUAL || assertionKind == OAK_NOT_EQUAL);
+    assert((assertionKind == OAK_EQUAL) || (assertionKind == OAK_NOT_EQUAL));
     assertionKind = (assertionKind == OAK_EQUAL) ? OAK_NOT_EQUAL : OAK_EQUAL;
 
-    if (op1op->OperIs(GT_LCL_VAR))
+    if (op1->OperIs(GT_LCL_VAR))
     {
-        return optCreateJtrueAssertions(op1, op2, assertionKind);
+        return optCreateJtrueAssertions(op1, op2, assertionKind, true);
```

Comment:
Yes, inverting the `if` sounds good.

Reply:
Speaking of my earlier comment about public `gtCallArgs`/`gtCallLateArgs` and potential future cleanup - you can see in this code an example of dubious and potentially unsafe complication - the arg nodes are obtained from `gtCallLateArgs` rather than `gtCallArgs`. And since the order in which they appear (assuming that they even appear in late args) is not quite guaranteed the code tries to swap them. Nedless to say that more reasonable code would have simply done something like:
```C++
op1 = call->GetArgNode(0);
op2 = call->GetArgNode(1);
```